### DISCUSSION TITLE
Ability to run financial reports across all Organizations

### DISCUSSION THREAD

**Discussion by adrianjean** [craftcms/console#119] [Craft Console]
**Context:** 
- We are a medium sized agency with many clients for whom we build CraftCMS websites.
- We maintain these websites on their behalf, this includes paying for licenses.
- Since having the ability to create organizations and move licenses into those organizations, I have been doing so for all of our clients to keep things organized and attach PO numbers to each account.
- Because we pay for all of our client's licenses as part of their SLA with us, each organization has the same payment information.

**Issue:**
- The Organization features is great and I love it, but it also limits my ability to run aggregated financial reports on how much in licenses we've spent in a specified amount of time, over all of the organizations we've created, for CraftCMS and/or for plugins.
- To run this report, I have to go into each organization and click on invoices, and record the payments, plugins, etc. in a spreadsheet and slice and dice from there.
- At the end of every quarter when we review our expenses, this is becomes bit of a gruelling task.

**Idea:**
- Would it be possible to have a Financial Reporting section that could list all invoices within a date range, filtered by CraftCMS 
 and/or Plugin, either grouped by organization and/or filter by organization with CSV exports? ... (I know it's a lot)

**Additional Improvements**
- In my "Personal" or individual primary account, when I select "Orders" from the sidebar, I get a listing of transactions with dates, values and invoice numbers.
- However inside an organization, clicking on "Orders" in the sidebar yields a different interface with far fewer details. (See attached screenshot)
- Would it be possible to have the Orders section have all the same details as the "Personal"/individual account orders page?
- Would also be nice to have an export to CSV button for these lists as well.

This would make it much easier to manage multiple websites with multiple licences over years, and track expenses, and recoup that revenue.

Right now, I'm contemplating moving all my licenses back into my user account (like it was before Organizations feature), but then I would lose the organization and separation of expenses I have now.

<img width="769" alt="Screenshot 2024-09-23 at 7 54 31 AM" src="https://github.com/user-attachments/assets/4c089aa0-4b60-4d9b-800a-40d45fb40915">

**Comment by timkelty** [comment]
Sounds like a very useful feature when you're fronting the costs for organization, for sure.

I'm curious how many people/agencies manage orgs like this, vs. using the client's billing details directly…

**Reply by adrianjean** [reply]
I know... this is why the Organization feature was created, to create separation! For that purpose it's fantastic. If we had a different model that directly charged clients for their licenses it would be a dream to use. 

But we strive create client relationships that don't focus on the minutia of billing and technical bits — we handle all that for them — yet at the same time, we love having licenses organized by client/project (...as some clients/projects have multiple websites).

So.... I worry this makes us a bit of an edge case.

**Comment by adrianjean** [comment]
I added a section on additional improvements. I noticed the Orders interface is different between an individual account and an organization. With could also help simplify / speed up accounting/reporting.

**Comment by ZapaNOR** [comment]
I was about to make a separate suggestion but saw this.
We are a medium / big agency and we front all the licenses for our clients, since most of them dont want to deal with it themself...

We started using the organization feature, but quickly saw that we got too many for it to be maintainable, since each org. needs a payment card, and maintaining the card across all orgs. is a big job whenever we need to change it.

Getting an overview was also hard, so now we have almost all licenses under our own org. so we can see the next renewal dates at a glance.

I would love for a feature where we get all orders or licenses in one place, maybe grouped by org for those who use that.
We would also love very much to be able to see the renewal **price** when looking at the Craft CMS license overview, since right now you have to click in on every one, and then you only see the CMS license, if you want to also know what the plugin costs are at renewal, you need to click each of those too. With 40+ sites atm, and quickly growing, this takes up a lot of time.

**Reply by timkelty** [reply]
Thanks Mathias – we'll try and come up with some solutions in the new year.

> since each org. needs a payment card, and maintaining the card across all orgs

That's a tough one, since having a separate card is part of the point of an org, but I understand how managing this would be a pain. 

Maybe the solution there could be a UI to bulk-update the card for any orgs you have permission to do so?

**Reply by ZapaNOR** [reply]
That could work, or when creating an org you could check off "Give this org access to primary payment card" or something like that. So they just get access to the one on the main account, but still has the ability to add another themself if they want. Then not all orgs. automatically sees the primary card.

Also, if I had to prioritize, beeing able to see upcoming renewal prices at a glance is way up higher than cards atm, luckily, we dont update them too often
